engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/PartSort.scala

```diff
@@ -0,0 +1,39 @@
+package pl.touk.nussknacker.engine.compile
+
+import pl.touk.nussknacker.engine.graph.node.{BranchEndData, BranchEndDefinition, SourceNodeData}
+import pl.touk.nussknacker.engine.splittedgraph.SplittedNodesCollector
+import pl.touk.nussknacker.engine.splittedgraph.part.SourcePart
+import pl.touk.nussknacker.engine.splittedgraph.splittednode.EndingNode
+
+import scala.annotation.tailrec
+
+object PartSort {
+
+  @tailrec
+  //this is primitive topological sort, we want to sort parts,
```

Comment:
Can we move it on the class level? And also write some details why this sorting is need? Or at least notice in the place of invocation...

Reply:
Added comment